esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1920,6 +1920,30 @@ static bool parseUNCPath(const char* sprayPath, StringBuffer& localPath, StringB
     return true;
 }
 
+static IPropertyTree* findDropZonePlaneForSpray(const char* hostReq, StringArray& files)
+{
+    SocketEndpoint hostEp(hostReq);
+    ForEachItemIn(i, files)
+    {
+        const char* file = files.item(i);
+        if (isEmptyString(file))
+            continue;
+
+        StringBuffer hostInPath, firstFilePath;
+        if (!parseUNCPath(file, firstFilePath, hostInPath))
+            firstFilePath.set(file);
+        else
+        {
+            SocketEndpoint hostInPathEp(hostInPath);
+            if (!hostEp.ipequals(hostInPathEp))
+                throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "The host '%s' defined in the SourcePath '%s' does not match with the host '%s' defined in SourceIP.", hostInPath.str(), file, hostReq);
+        }
+        if (isAbsolutePath(firstFilePath))
+            return findDropZonePlane(firstFilePath, hostReq, true, isContainerized());
```

Comment:
I found that the filename.csv does not have to be stripped because the findDropZonePlane() will call the isPathInPlane() to validate the path:
bool isPathInPlane(IPropertyTree *plane, const char *path)
{
    return isEmptyString(path) || startsWith(path, plane->queryProp("@prefix"));
}

Reply:
> I know it's not supposed to / not supported, if they have multiple filenames that point to different planes, but it would be validating that here, i.e. don't just return 1st match. Instead retain it, and check other paths match, or if a UNC, both the host and the path match - fire an error if any don't (unsupported).

The existing code does check other paths and host in line 2001 to 2039. Do we want to duplicate the function or refactor them in this PR?

Reply:
> I found that the filename.csv does not have to be stripped because the findDropZonePlane() will call the isPathInPlane() to validate the path: bool isPathInPlane(IPropertyTree *plane, const char *path) { return isEmptyString(path) || startsWith(path, plane->queryProp("@Prefix")); }

ok, yes agree, I misinterpreted what isPlaneInPath was doing. It's fine as it is.

Reply:
> The existing code does check other paths and host in line 2001 to 2039. Do we want to duplicate the function or refactor them in this PR?

it might functionally do what we want, but findDropZonePlaneForSpray is misleading and would confuse if for example it were reused and not spot the inconsistencies.

It would be better to merge this validation into the existing 'files' loop.
i.e. don't call findDropZonePlane upfront, but instead call it in the loop 'dropZone' is not set, then all subsequent iterations around the loop can validate against it (as it already does).

Reply:
Let's say there are 4 filenames. Only the 3rd one has an absolute path which can be used to find out the dropzone and the prefix. Should the code go back to the 1st filename and 2nd filename to set the absolute path, check FileScopePermission, and sourcePathReq?

Reply:
Inside the existing findDropZonePlaneForSpray(), if the dropzone is not found based on an absolute path (bare metal + isDropZoneRestrictionEnabled()==false) after the 3rd filename. If the same situation happens after merging this validation into the existing 'files' loop, should the code try to find out the dropzone if the 4th filename has an absolute path? If the dropzone is found, what should be done for the 1st, 2nd, and 3rd filename?

Reply:
The code for checking FileScopePermission and building absolute path may be moved out of the existing 'files' loop.

Reply:
I would not repeatedly lookup a dropzone.
Lookup based on the 1st file, if it is relative, then it has to lookup by host only.
All subsequent files the csv list, should validate they correspond go the same dropzone.
i.e. if any are absolute, verify it is a path within the established dropzone, if any are UNC, verify that both host and path are in the established dropzone.
If any do not match - then there's inconsistency and this is not supported, it should throw an exception

Reply:
> The code for checking FileScopePermission and building absolute path may be moved out of the existing 'files' loop.

I think you can remove findDropZonePlaneForSpray, and fold in the logic into the loop, checking if dropZone is not set (i.e. will not be set, unless sourcePlaneReq is supplied, on 1st iteration).
Then have code like this under the parseUNCPath block:
```
                if (!dropZone)
                {
                    const char *dropZonePath = isAbsolutePath(hostInPath) ? hostInPath.str() : nullptr;
                    dropZone.setown(findDropZonePlane(dropZonePath, hostInPath, true, isContainerized()));
                }
                else
                {
                    // don't re-check if sourceIPReq specified and therefore already checked valid previously.
                    .....
```

and similar under the non UNC block.

Reply:
all inconsistencies should be rejected.
Find the dropzone based on the 1st file + IP from UNC, or IP from request, and if any subsequent ones do not lie in that dropzone, throw an error.

Reply:
ok

Reply:
For bare-metal and isDropZoneRestrictionEnabled()==false, the findDropZonePlane() may return nullptr. I guess the code should call the findDropZonePlane() if it is 1st file and !dropZone?